monika/linux/generic_syscall.cpp

```diff
@@ -0,0 +1,58 @@
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include "debugbreak.h"
+#include <string.h>
+#include <haiku_errors.h>
+#include <BeDefs.h>
+#include "linux_syscall.h"
+
+#define RANDOM_SYSCALLS "random"
+#define RANDOM_GET_ENTROPY 1
+struct random_get_entropy_args {
+	void *buffer;
+	size_t length;
+};
+static inline int _open(const char *hostPath, int mode) {
+	return LINUX_SYSCALL4(__NR_openat, AT_FDCWD, hostPath, 0, mode);
+}
+static inline ssize_t _read(int fd, void *buffer, size_t bufferSize) {
+	return LINUX_SYSCALL3(__NR_read, fd, buffer, bufferSize);
+}
+static inline int _close(int fd) {
+        return LINUX_SYSCALL1(__NR_close, fd);
+}
+extern "C" {
+int32_t _moni_generic_syscall(const char *userSubsystem, uint32_t function, void *buffer, uint64_t bufferSize) {
+	char subsystem[B_FILE_NAME_LENGTH];
+	if (!strcmp(userSubsystem, RANDOM_SYSCALLS)) {
+		switch (function) {
+			case RANDOM_GET_ENTROPY: {
+				random_get_entropy_args args;
+				if (bufferSize != sizeof(args)) {
+					return B_BAD_VALUE;
+				}
+				memcpy(&args, buffer, sizeof(args));
+				int random_fd = _open("/dev/random", O_RDONLY);
```

Comment:
I would suggest using the [getrandom](https://www.man7.org/linux/man-pages/man2/getrandom.2.html) Linux syscall to avoid polluting the FD table.